metaparser/xcarchive.go

```diff
@@ -0,0 +1,76 @@
+package metaparser
+
+import (
+	"fmt"
+
+	"github.com/bitrise-io/go-xcode/v2/artifacts"
+	"github.com/bitrise-io/go-xcode/v2/zip"
+)
+
+// ParseXCArchiveData ...
+func (m *Parser) ParseXCArchiveData(pth string) (*ArtifactMetadata, error) {
+
+	appInfo, scheme, err := m.readXCArchiveDeploymentMeta(pth)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse deployment info for %s: %w", pth, err)
+	}
+
+	fileSize, err := m.fileManager.FileSizeInBytes(pth)
+	if err != nil {
+		m.logger.Warnf("Failed to get apk size, error: %s", err)
+	}
+
+	return &ArtifactMetadata{
+		AppInfo:       appInfo,
+		FileSizeBytes: fileSize,
+		Scheme:        scheme,
+	}, nil
+}
+
+func (m *Parser) readXCArchiveDeploymentMeta(pth string) (Info, string, error) {
+	reader, err := zip.NewDefaultReader(pth, m.logger)
+	if err != nil {
+		return Info{}, "", err
+	}
+	defer func() {
+		if err := reader.Close(); err != nil {
+			m.logger.Warnf("%s", err)
+		}
+	}()
+
+	xcarchiveReader := artifacts.NewXCArchiveReader(reader)
+	isMacos := xcarchiveReader.IsMacOS()
+	if isMacos {
+		m.logger.Warnf("macOS archive deployment is not supported, skipping xcarchive")
```

Comment:
I think in this lib, this should be an error and the client should decide how to handle this situation.